### DISCUSSION TITLE
How do I get a file as a String/&str?

### DISCUSSION THREAD

**Discussion by beyarkay** [XAMPPRocky/octocrab#510] [Q&A]
I want to get https://github.com/beyarkay/eskom-calendar/blob/main/manually_specified.yaml as a `String` or `&str` so that I can parse the YAML and edit it. How do I do this? I've tried:

```rust
let body = octocrab::instance()
    .repos("beyarkay", "eskom-calendar")
    .raw_file("main".to_string(), "manually_specified.yaml")
    .await
    .unwrap()
    .into_body();
```

And that gets me a `hyper::Body` object, but I can't figure out how to extract the content from it. I've tried to look online, but all the answers [1](https://stackoverflow.com/questions/38148163/displaying-the-response-body-with-hyper-only-shows-the-size-of-the-body) [2](https://users.rust-lang.org/t/convert-hyper-body-to-string/23179) [3](https://users.rust-lang.org/t/extracting-hyper-request-body-into-a-string/12723) [4](https://stackoverflow.com/questions/63301838/how-to-read-the-response-body-as-a-string-in-rust-hyper) suggest things that don't work with the latest version of hyper.

Is it possible to get the raw text of a file hosted on github? I've resorted to using `reqwest` and just downloading the raw.githubusercontent.com version of the file, but the `raw_file` method *sounds* like it should be able to give me the raw file.

**Comment by landaire** [comment]
Old issue but I was able to figure it out:

```rust
 use http_body::Body;

if let Ok(constants_updates) = octocrab
    .repos("foo", "bar")
    .raw_file(Reference::Branch("main".to_string()), "data/latest.json")
    .await
{
    let mut body = constants_updates.into_body();
    let mut result = Vec::with_capacity(body.size_hint().exact().unwrap_or_default() as usize);

    // Iterate through all data chunks in the body
    while let Some(frame) = body.frame().await {
        match frame {
            Ok(frame) => {
                if let Some(data) = frame.data_ref() {
                    result.extend_from_slice(&data);
                }
            }
            Err(_) => return None,
        }
    }

    Some(result))
} else {
    None
}
```
